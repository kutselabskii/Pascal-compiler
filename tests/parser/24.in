program abc;

var a: integer;

begin
repeat
	a := a + 1;
	a := a + 2;
	a := 14 mod 7;
until (a > 0);
end.