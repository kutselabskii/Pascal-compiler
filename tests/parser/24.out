|__ program
    |-- abc //Program_heading
    |__ //Block
        |-- //Declaration part
        |   |-- var
        |   |   |-- a
        |   |   |   |__  : integer
        |__ //Statement part
            |__ //Statement sequence
                |-- repeat
                |   |-- //Statement sequence
                |   |   |-- //Assignment_statement
                |   |   |   |-- a
                |   |   |   |   |__  : integer
                |   |   |   |__ addition
                |   |   |       |-- a
                |   |   |       |   |__  : integer
                |   |   |       |__ 1
                |   |   |__ //Assignment_statement
                |   |       |-- a
                |   |       |   |__  : integer
                |   |       |__ addition
                |   |           |-- a
                |   |           |   |__  : integer
                |   |           |__ 2
                |   |   |__ //Assignment_statement
                |   |       |-- a
                |   |       |   |__  : integer
                |   |       |__ 0
                |   |__ strictly_greater_than
                |       |-- a
                |       |   |__  : integer
                |       |__ 0
